include/swift/AST/DiagnosticsClangImporter.def

```diff
@@ -231,7 +231,11 @@ ERROR(foreign_reference_types_invalid_retain_release, none,
       "type '%2'",
       (bool, StringRef, StringRef))
 
-ERROR(foreign_reference_types_retain_release_non_void_return_type, none,
+ERROR(foreign_reference_types_retain_non_void_or_self_return_type, none,
```

Comment:
If one of these only apply for retain and the other for release, we could get rid of the select in both.